volmdlr/wires.py

```diff
@@ -3717,20 +3712,34 @@ def split(self, split_start, split_end):
                 volmdlr.edges.Arc2D(split_start, interior_point2,
                                     split_end)]
 
-    def discretise(self, n: float) -> List[volmdlr.Point2D]:
-        points = []
-        vector = volmdlr.Vector2D(0, 1)
-
-        for i in range(n):
-            points.append(self.center + self.radius * vector)
-            vector.rotation(center=volmdlr.Point2D(0, 0), angle=2 * math.pi / n, copy=False)
-            vector.normalize()
-
-        return points
-
-    def polygon_points(self, angle_resolution=10):
-        return volmdlr.edges.Arc2D.polygon_points(
-            self, angle_resolution=angle_resolution)
+    # def discretise(self, n: float):
+    #     # BUGGED: returns method
+    #     circle_to_nodes = {}
+    #     nodes = []
+    #     if n * self.length() < 1:
+    #         circle_to_nodes[self] = self.border_points
+    #     else:
+    #         n0 = int(math.ceil(n * self.length()))
+    #         l0 = self.length() / n0
+    #
+    #         for k in range(n0):
+    #             node = self.point_at_abscissa(k * l0)
+    #
+    #             nodes.append(node)
+    #
+    #         circle_to_nodes[self] = nodes
+    #
+    #     return circle_to_nodes[self]
+
+    def dicretization_points(self, angle_resolution=10):
+        return volmdlr.edges.Arc2D.discretization_points(
+            self, discretization_resolution=angle_resolution)
```

Comment:
ahh, Why do you think so? If I remember it correctly, only the method's name changed from polygon_points to discretisation_points

Reply:
I'm surprised it works... because you're using `Arc2D.discretization_points` with the `self` of a `Circle2D`, and there is no inheritance between these two. 
It works just because `Circle2D` has an attribute named `angle`, and two methods named `length` and `point_at_abscissa`. However, there was no guarantee this would work. 
The right way to do it is : 
```suggestion
    def dicretization_points(self, angle_resolution=10):
        return volmdlr.edges.Arc2D(start=?, interior=?, end=?, name=?).discretization_points(
            discretization_resolution=angle_resolution)
```

Reply:
hmmm. I see what you mean. You're indeed right, it is not correct. Paying close attention to the solution you have proposed, I thought it would be more costly to try to find the start, interior and end points, and also do all the calculations inside Arc2D init method, to just calculate the discretisation points... then I tried to reuse the method already developed, by a Mixin class to store this method and make Arc2D, and Circle2D inherit from it. What do you think, @AxelRinghausen ?